manticore/native/memory.py

```diff
@@ -892,6 +892,19 @@ def perms(self, index):
         else:
             return self.map_containing(index).perms
 
+    def max_exec_size(self, index, size):
+        r = 0
+        addr = index
+        while addr < index + size:
+            if addr not in self:
+                return r
+            m = self.map_containing(addr)
+            if not m.access_ok("x"):
```

Comment:
Heh, this uses `.perm` attribute under the hood which is public... Maybe this attribute should be private? Or maybe we should use it instead and do `if 'x' in m.perm`. We should probably leave it as it is for now and keep that in mind/possibly get back to it in the future.

Reply:
ok, adding a comment